File: answers/leetcode/Gray Code.cpp
class Solution {
public:
	vector<int> grayCode(int n) {
		vector<int> ans;
		ans.push_back(0);
		for (int i = 0; i < n; ++i) {
			int highest_bit = 1 << i;
			int len = ans.size();
			for (int j = len - 1; j >= 0; --j) {
				ans.push_back(highest_bit + ans[j]);
			}
		}
		return ans;
	}
};

File: answers/leetcode/Gray Code/Gray Code.java
class Solution {
    public List<Integer> grayCode(int n) {
        List<Integer> result = new ArrayList<>();
        result.add(0);
        for (int i = 0; i < n; i++) {
            int size = result.size();
            int bit = 1 << i;
            for (int j = size - 1; j >= 0; j--) {
                result.add(result.get(j) | bit);
            }
        }
        return result;
    }
}